space_packet_parser/xtce/definitions.py

```diff
@@ -40,82 +46,81 @@ class XtcePacketDefinition(common.AttrComparable):
+            XML namespace mapping, expected as a dictionary with the keys being namespace labels and
+            values being namespace URIs. Default {DEFAULT_XTCE_NSMAP}. An empty dictionary indicates no namespace
+            awareness, in which case `xtce_ns_prefix` must be None.
+        xtce_ns_prefix : str
+            XTCE namespace prefix. Default {DEFAULT_XTCE_NS_PREFIX}. This is the key for the XTCE namespace in the
+            namespace mapping dictionary, `ns` and is used to write XML output when necessary.
         root_container_name : Optional[str]
             Name of root sequence container (where to start parsing)
         space_system_name : Optional[str]
             Name of space system to encode in XML when serializing.
+        validation_status : str
+            One of ["Unknown", "Working", "Draft", "Test", "Validated", "Released", "Withdrawn"].
         xtce_version : str
             Default "1.0"
         date: Optional[str]
             Optional header date string.
         """
+        if xtce_ns_prefix and xtce_ns_prefix not in ns:
+            raise ValueError(f"XTCE namespace prefix {xtce_ns_prefix} not in namespace mapping {ns}. This is an error.")
+
         self.parameter_types = {}
         self.parameters = {}
         self.containers = {}
 
         def _update_caches(sc: containers.SequenceContainer) -> None:
-            if sc.name in self.containers and self.containers[sc.name] is not sc:
-                raise ValueError(f"Got two different SequenceContainer objects: "
-                                 f"{sc}, {self.parameter_types[sc.name]}, "
-                                 f"for SequenceContainer with name {sc.name}. If you reuse a "
-                                 f"SequenceContainer, you must pass in the same object for each instance. Note: "
-                                 f"these SequenceContainer objects may be __eq__ equal but they "
-                                 f"are not the same object ")
+            """Iterate through a SequenceContainer, updating internal caches with all Parameter, ParameterType,
+            and SequenceContainer objects, ensuring that a key (object name) only references a single object.
+
+            Notes
+            -----
+            This catches cases where, e.g. a Parameter element has been parsed twice, resulting in two Parameter
+            objects which are "equal" but not the same memory reference.
+
+            Parameters
+            ----------
+            sc : containers.SequenceContainer
+                The SequenceContainer to iterate through.
+            """
             self.containers[sc.name] = sc
             for entry in sc.entry_list:
                 if isinstance(entry, containers.SequenceContainer):
                     _update_caches(entry)  # recurse
                 elif isinstance(entry, parameters.Parameter):
-                    if entry.name in self.parameters and self.parameters[entry.name] is not entry:
```

Comment:
This is just overwriting the previous object with the new one now it looks like. Do you want to warn/log instead of raising still?

Reply:
I opted to just remove those checks entirely. I do want it to be the expectation but those lookups were eating up a lot of time in the XtcePacketDefinition creation benchmark. 

I think I'd rather have it encapsulated in a validation method that can be turned off if speed is important. Maybe I'll take a stab at a basic validation method for object references.